### PR TITLE
Integrate archive loading into test-e2e-cip-auditor

### PR DESCRIPTION
**Pull request by tylerferrara**
#### What type of PR is this?
/kind cleanup
/kind deprecation

#### What this PR does / why we need it:
Remove the bazel build and push process for golden images, in favor of archive loading and pushing. This simply calls the existing `push-golden.sh` script to handle this process completely. However, since the archive images are being used for testing, existing fixtures must be updated to reflect the new sha256 digests.

Partially satisfies #304

#### Which issue(s) this PR fixes:
None

#### Does this PR introduce a user-facing change?

```release-note
Integrate push-golden.sh into cip-auditor-e2e.go.
Change e2e auditor test fixtures to reflect golden archives.
```
cc: @listx @amwat @justaugustus @kubernetes-sigs/release-engineering

### CODE REVIEW

**Comment by tylerferrara** [comment]
**Blocked** by [#22658](https://github.com/kubernetes/test-infra/pull/22658) and #314

**Comment by listx** [comment]
/lgtm
/approve
/hold

Holding for all presubmits to pass.

File: test-e2e/cip-auditor/cip-auditor-e2e.go [hunk 1]
```diff
@@ -299,96 +299,28 @@ func testSetup(
...
 
-	if err := populateGoldenImages(repoRoot, pushRepo); err != nil {
+	if err := populateGoldenImages(repoRoot); err != nil {
 		return err
 	}
 
 	return nil
 }
 
 // nolint[funlen]
```
Comment (justaugustus) [inline test-e2e/cip-auditor/cip-auditor-e2e.go:309]: Non-blocking: let's drop this in a follow-up.

**Review by justaugustus — APPROVED** [review]

File: test-e2e/cip-auditor/cip-auditor-e2e.go (hunk 1 above)
Comment (listx) [reply test-e2e/cip-auditor/cip-auditor-e2e.go:309]: Nice catch.

Comment (tylerferrara) [reply test-e2e/cip-auditor/cip-auditor-e2e.go:309]: Great catch! Changing now.

**Pull request merged**